### DISCUSSION TITLE
S3 via Access Keys

### DISCUSSION THREAD

**Discussion by robserva** [twentyhq/twenty#5072] [Engineering vision]
It's definitely possible I'm just missing something, but it looks like Twenty doesn't currently support providing an access key ID and secret for S3 compatible storage. I'm using fly.io with Tigris, so an IAM based solution doesn't work.

Does this seem desirable for the engineering plans?

**Comment by FelixMalfait** [comment]
@robserva if a PR is open for this that's something we would be happy to merge yes! Probably not that big of a change?

**Comment by janbiasi** [comment]
@robserva I'm not sure if I understand/interpret your input correctly, but it is already possible to authenticate via access key id and secret access key, see [file-storage.module-factory.ts (impl.)](https://github.com/twentyhq/twenty/blob/5ebde33f5f73626188cc110c99374dce176e7fb7/packages/twenty-server/src/engine/integrations/file-storage/file-storage.module-factory.ts#L31-L57). Just make sure to set the corresponding [environment variables](https://twenty.com/developers/section/self-hosting/self-hosting-var#storage) in your `docker-compose.yml` file and hosting provider.

**Reply by FelixMalfait** [reply]
Ah yes someone kindly opened a PR for this and we forgot to reference it in the discussion. Closing this now. Thanks!
